### ISSUE TITLE
Plugin Crash

### ISSUE THREAD

**Issue by RaptorPharaoh** [FriendsOfGalaxy/galaxy-integration-steam#63]
**Describe the bug**
After v0.46.5 the plugin not work fine for me, all the time I have errors or disconnections. Including the plugin not work at all. I have to install the v0.46.5 or v0.46 to work again

**Comment by FriendsOfGalaxy** [comment]
@RaptorPharaoh  Please provide your logs from C:\ProgramData\GOG.com\Galaxy\logs folder

**Comment by RaptorPharaoh** [comment]
Hello, I have a lot of them and I tested a lot, but I think this is the one failed
[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4628688/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

I put the v0.46 and work perfectly. Without any problem.

**Comment by derekhansell** [comment]
I also had the same problem with the version being downloaded automatically by GOG Galaxy, and fixed it by reverting back to version 0.46. You should know there are a bunch of angry folks over on the GOG Galaxy forums complaining about this. I pointed them here.

https://www.gog.com/forum/general_beta_gog_galaxy_2.0/steam_plugin_crashing_constantly

**Comment by RaptorPharaoh** [comment]
> I also had the same problem with the version being downloaded automatically by GOG Galaxy, and fixed it by reverting back to version 0.46. You should know there are a bunch of angry folks over on the GOG Galaxy forums complaining about this. I pointed them here.
> 
> https://www.gog.com/forum/general_beta_gog_galaxy_2.0/steam_plugin_crashing_constantly

For a moment I thought it was just me :(

**Comment by FriendsOfGalaxy** [comment]
@RaptorPharaoh  i dont see any error in your logs :( can you provide the whole folder ? here or on my email friendsofgalaxy@gmail.com

**Comment by RaptorPharaoh** [comment]
> @RaptorPharaoh i dont see any error in your logs :( can you provide the whole folder ? here or on my email [friendsofgalaxy@gmail.com](mailto:friendsofgalaxy@gmail.com)

@FriendsOfGalaxy Hello, I sent you an Email with the new logs focused in the crash/fail :)

**Comment by leebailey88** [comment]
Also happening for me

**Comment by InsanityOnABun** [comment]
Running into this regularly as well.

**Comment by TicklesTheBrain** [comment]
Crashes all the time.

**Comment by FriendsOfGalaxy** [comment]
@leebailey88 @InsanityOnABun @TicklesTheBrain 
Please send logs, saying 'its crashing' without the logs doesn't help with anything.

You can find the logs here C:\ProgramData\GOG.com\Galaxy\logs ,  you can send them here or on my mail  friendsofgalaxy@gmail.com

**Comment by InsanityOnABun** [comment]
@FriendsOfGalaxy there a quite a few logs in that folder, which ones do you need? Just the steam plugin log? Several others? All of them? Please clarify.

**Comment by FriendsOfGalaxy** [comment]
@InsanityOnABun  Steam plugin logs will be sufficient. Their naming conventions is 
plugin-steam-***.log

**Comment by TicklesTheBrain** [comment]
[logs.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4651762/logs.zip)
Find logs attached. They are quite bit, I've got loads of stuff on steam, sorry for that.

**Comment by InsanityOnABun** [comment]
[logs.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4654494/logs.zip)
See link for logs.

**Comment by p-smith** [comment]
@FriendsOfGalaxy The plugin pretty often crashes for me too. Interestingly the majority of the crashes occur when a particular friend of mine is online (I observed several times that the log suddenly stops during importing user presence, and always at steam user '76561197983387131'). When he's offline, the plugin is working fine. Here are the logs if you want to take a look:
[logs.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4658627/logs.zip)

**Comment by FriendsOfGalaxy** [comment]
@p-smith  this is huge help! I will work on it asap, thanks!

**Comment by FriendsOfGalaxy** [comment]
@p-smith  Can you also send some of your GalaxyClient.logs?

here or on my email friendsofgalaxy@gmail.com i want to check how galaxy behaves during this crash, maybe it has some information which plugin log doesnt.

**Comment by p-smith** [comment]
@FriendsOfGalaxy Sure. I'm at work, so sadly I don't have access to the logs right now, but I can send you the extra logs in the evening (CEST). Sorry for the delay.

**Comment by FriendsOfGalaxy** [comment]
@p-smith can you take a look at this version and see if the plugin still crashes for you ? 
[windows.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4662093/windows.zip)

 To install it 
0 close Galaxy Client
1 delete the contents of your C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
2 unpack the contents of windows.zip folder to C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
3 run Galaxy Client

I've basically added a timeout to rich presence translation so it wont recurse forever and added some additional logs which will help me debug the issue if it occurs again

**Comment by janitorr** [comment]
> 
> 
> @p-smith can you take a look at this version (after you get back from work of course 😄 ) and see if the plugin still crashes for you ?
> [windows.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4662093/windows.zip)
> 
> To install it
> 0 close Galaxy Client
> 1 delete the contents of your C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
> 2 unpack the contents of windows.zip folder to C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
> 3 run Galaxy Client
> 
> I've basically added a timeout to rich presence translation so it wont recurse forever and added some additional logs which will help me debug the issue if it occurs again

I was suffering from same issue and did this just a moment ago, still monitoring but at least not crashing as soon as before.

**Comment by FriendsOfGalaxy** [comment]
> > @p-smith can you take a look at this version (after you get back from work of course 😄 ) and see if the plugin still crashes for you ?
> > [windows.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4662093/windows.zip)
> > To install it
> > 0 close Galaxy Client
> > 1 delete the contents of your C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
> > 2 unpack the contents of windows.zip folder to C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
> > 3 run Galaxy Client
> > I've basically added a timeout to rich presence translation so it wont recurse forever and added some additional logs which will help me debug the issue if it occurs again
> 
> I was suffering from same issue and did this just a moment ago, still monitoring but at least not crashing as soon as before.

Awesome @janitorr  in case you run into the issue be sure to send me your logs, i'm not exactly sure how many people are affected by plugin getting stuck at importing rich presence (since its a different issue from err 5 which is most crashes i've seen here)

**Comment by janitorr** [comment]
> 
> 
> > > @p-smith can you take a look at this version (after you get back from work of course 😄 ) and see if the plugin still crashes for you ?
> > > [windows.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4662093/windows.zip)
> > > To install it
> > > 0 close Galaxy Client
> > > 1 delete the contents of your C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
> > > 2 unpack the contents of windows.zip folder to C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
> > > 3 run Galaxy Client
> > > I've basically added a timeout to rich presence translation so it wont recurse forever and added some additional logs which will help me debug the issue if it occurs again
> > 
> > 
> > I was suffering from same issue and did this just a moment ago, still monitoring but at least not crashing as soon as before.
> 
> Awesome @janitorr in case you run into the issue be sure to send me your logs, i'm not exactly sure how many people are affected by plugin getting stuck at importing rich presence (since its a different issue from err 5 which is most crashes i've seen here)

I was just just keeping it open for an hour and playing a game launch from GOGG for another hour and not a single crash. I think this did it for me.

**Comment by p-smith** [comment]
> @p-smith can you take a look at this version (after you get back from work of course 😄 ) and see if the plugin still crashes for you ?
> [windows.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4662093/windows.zip)
> 
> To install it
> 0 close Galaxy Client
> 1 delete the contents of your C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
> 2 unpack the contents of windows.zip folder to C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
> 3 run Galaxy Client
> 
> I've basically added a timeout to rich presence translation so it wont recurse forever and added some additional logs which will help me debug the issue if it occurs again

I've installed this version too, so far so good, no crashes yet. I'll be monitoring GOGG and let you know if something happens.

**Comment by p-smith** [comment]
Like @janitorr I kept GOGG open, played a few games and the plugins is as stable as ever. I think this did it for me too.

**Comment by FriendsOfGalaxy** [comment]
Released plugin version 0.50.8, it should help for the crash on rich presence issue as well as help any people that were recently affected by no owned games showing issue.

Still working on err 5!

**Comment by Varstahl** [comment]
It wasn't working properly for me either, then I've updated to the latest 0.50.8 and it started crashing immediately. Seems like an issue with the imports:
```
ImportError: cannot import name 'etree' from 'lxml' (C:\Users\Varstahl\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\lxml\__init__.py)
2020-06-01 00:34:40,215 - root - ERROR - Plugin exception
Traceback (most recent call last):
  File "C:\Program Files (x86)\GOG Galaxy\plugin_runner.py", line 35, in <module>
    module = importlib.import_module(filename)
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "C:\Users\Varstahl\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\plugin.py", line 22, in <module>
    from backend import SteamHttpClient, AuthenticatedHttpClient, UnfinishedAccountSetup
  File "C:\Users\Varstahl\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\backend.py", line 10, in <module>
    from requests_html import HTML
  File "C:\Users\Varstahl\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\requests_html.py", line 11, in <module>
    from pyquery import PyQuery
  File "C:\Users\Varstahl\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\pyquery\__init__.py", line 7, in <module>
    from .pyquery import PyQuery  # NOQA
  File "C:\Users\Varstahl\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8\pyquery\pyquery.py", line 11, in <module>
    from lxml import etree
```
Logs, just in case you might need them: [plugin-steam-logs.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4708640/plugin-steam-logs.zip)

**Comment by BuryTomorrow12** [comment]
Here we go again.

**Version:** 0.51.1
**Issue:** Steam plugin crashes 15 seconds after connecting my steam account
**Steam guard enabled:** Yes
**Steps to reproduce:** Do a clean install of 0.51.1 (removed previous version, removed all steam plugin files, deleted logs, copied new plugin files). Then relaunch galaxy, reconnect steam plugin, manually typing credentials. 15 seconds later, plugin crashes.

Log attached.

[plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/4794627/plugin-steam-ca27391f-2675-49b1-92c0-896d43afa4f8.log)

**Comment by Qubits01** [comment]
I have the same problem with 0.51.1

**Comment by Sleazus** [comment]
TLDR: try adding a different integration and see if that helps importing your games. May need to add 2, or disconnect and reconnect the other integration. Seems to import games even when crashed. Imported games stay with you even on a clean windows/gog install without being connected.

Hey, So I've been having amazing troubles getting all my steam games to be visible on gog.
What has worked for me as a work around was connecting a different integration (i connected rockstar for example)
Gog crashed. I had to do several restarts and connect another integration but it seems to have most if not all of my steam games there now.
Funnily enough it still imported even though the steam plugin had crashed.
It still crashes but yeah, my games seem to be in Gog.
I tried this awhile ago with a different version with a different computer and it kept crashing alot etc. but then i got a new computer, did a clean install, installed Gog and my games were still there despite not being connected.

Was using 51.1 for this. Not sure what I was using before

**Comment by FriendsOfGalaxy** [comment]
@Varstahl the error you pasted has appeared on plugin shutdown - something strange has happen for sure but it is not important for now. The real problem you have is #74 

@BuryTomorrow12 thanks, your trace is very helpful. The crash reason in your case is plugin is stuck for more than 5 seconds processing your games licenses. You have too many games. You have to refund some. Or try the the version from this comment: 
https://github.com/FriendsOfGalaxy/galaxy-integration-steam/issues/78#issuecomment-679073023

**Comment by BuryTomorrow12** [comment]
Thank you for pointing me out to that other version of the plugin, but are you really telling me I have to refund games I purchased so I can use a plugin for a gaming app that's not even the mandatory one to play said games? Wow. I've seen some replies but this one is just on another level.

**Comment by FriendsOfGalaxy** [comment]
@BuryTomorrow12 no! It was a joke, it was of course the plugin fault to not being optimized to handle bigger libraries :)

**Comment by derekhansell** [comment]
> @BuryTomorrow12 no! It was a joke, it is of course the plugin fault to not being optimized to handle bigger libraries :)

Whoops. Someone forgot to write /s on their joke that was clearly a joke. :)

**Comment by FriendsOfGalaxy** [comment]
Update: release candidate version of the plugin:
[steam_4_fixes.zip](https://github.com/FriendsOfGalaxy/galaxy-integration-steam/files/5140912/steam_4_fixes.zip)

To install it:
0 close Galaxy Client
1 delete the contents of your C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
2 unpack the contents of windows.zip folder to C:\Users\Username\AppData\Local\GOG.com\Galaxy\plugins\installed\steam_ca27391f-2675-49b1-92c0-896d43afa4f8 folder
3 run Galaxy Client

**Comment by FriendsOfGalaxy** [comment]
Hello, the fix was released some time ago. @RaptorPharaoh has your issue gog?
